flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/checkpointing/AlternatingController.java

```diff
@@ -35,15 +39,23 @@
 public class AlternatingController implements CheckpointBarrierBehaviourController {
     private final AlignedController alignedController;
     private final UnalignedController unalignedController;
+    private final DelayedActionRegistration delayedActionRegistration;
+    private final Clock clock;
 
     private CheckpointBarrierBehaviourController activeController;
-    private long firstBarrierArrivalTime = Long.MAX_VALUE;
+    private long lastBarrierArrivalTime = Long.MAX_VALUE;
```

Comment:
The meaning of this variable is: arrival time of the first barrier of the last seen checkpoint :)
So both old `firstBarrierArrivalTime` and new `lastBarrierArrivalTime` don't reflect it fully.
But to me, the old name is less confusing.